### PR TITLE
[DI] Fix tracking of env vars in exceptions

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q             | A
| ------------- | ---
| Branch?       | 3.3
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #24270, #24368
| License       | MIT
| Doc PR        | -

Fixes the bad exception message reported in the linked issue (regression introduced in v3.3.7).

Best reviewed [ignoring whitespaces](https://github.com/symfony/symfony/pull/25163/files?w=1).

### CODE REVIEW

File: src/Symfony/Component/DependencyInjection/Compiler/MergeExtensionConfigurationPass.php [hunk 1]
```diff
@@ -52,18 +52,26 @@ public function process(ContainerBuilder $container)
...
-            $tmpContainer->setResourceTracking($container->isTrackingResources());
-            $tmpContainer->addObjectResource($extension);
-            if ($extension instanceof ConfigurationExtensionInterface && null !== $configuration = $extension->getConfiguration($config, $tmpContainer)) {
-                $tmpContainer->addObjectResource($configuration);
-            }
+            try {
+                $tmpContainer = new ContainerBuilder($resolvingBag);
+                $tmpContainer->setResourceTracking($container->isTrackingResources());
+                $tmpContainer->addObjectResource($extension);
+                if ($extension instanceof ConfigurationExtensionInterface && null !== $configuration = $extension->getConfiguration($config, $tmpContainer)) {
```
Comment (SCIF) [inline src/Symfony/Component/DependencyInjection/Compiler/MergeExtensionConfigurationPass.php]: Assignment in condition?

File: src/Symfony/Component/DependencyInjection/Compiler/MergeExtensionConfigurationPass.php [hunk 2]
```diff
@@ -52,18 +52,26 @@ public function process(ContainerBuilder $container)
...
-            foreach ($exprLangProviders as $provider) {
-                $tmpContainer->addExpressionLanguageProvider($provider);
-            }
+                foreach ($exprLangProviders as $provider) {
+                    $tmpContainer->addExpressionLanguageProvider($provider);
+                }
 
-            $extension->load($config, $tmpContainer);
+                $extension->load($config, $tmpContainer);
+            } catch (\Exception $e) {
```
Comment (SCIF) [inline src/Symfony/Component/DependencyInjection/Compiler/MergeExtensionConfigurationPass.php]: Any chance to specify less general exception?

Comment (nicolas-grekas) [reply src/Symfony/Component/DependencyInjection/Compiler/MergeExtensionConfigurationPass.php]: Would be a bug. Could be "finally" in fact.

File: src/Symfony/Component/DependencyInjection/Compiler/MergeExtensionConfigurationPass.php (hunk 1 above)
Comment (nicolas-grekas) [reply src/Symfony/Component/DependencyInjection/Compiler/MergeExtensionConfigurationPass.php]: Indeed :) Was here before, and not forbidden by our policy, you can find many others in the code base.

**Pull request merged**